qiskit/qasm3/__init__.py

```diff
@@ -11,34 +11,126 @@
 # that they have been altered from the originals.
 
 """
-==========================
-Qasm (:mod:`qiskit.qasm3`)
-==========================
+================================
+OpenQASM 3 (:mod:`qiskit.qasm3`)
+================================
 
 .. currentmodule:: qiskit.qasm3
 
-.. autosummary::
-    :toctree: ../stubs/
+Qiskit provides some tools for converting between `OpenQASM 3 <https://openqasm.com>`__
+representations of quantum programs, and the :class:`.QuantumCircuit` class.  These will continue to
+evolve as Qiskit's support for the dynamic-circuit capabilities expressed by OpenQASM 3 increases.
 
-    Exporter
-    dumps
-    dump
+
+Exporting to OpenQASM 3
+=======================
+
+The high-level functions are simply :func:`dump` and :func:`dumps`, which respectively export to a
+file (given as a filename) and to a Python string.
+
+.. autofunction:: dump
+.. autofunction:: dumps
+
+Both of these exporter functions are single-use wrappers around the main :class:`Exporter` class.
+For more complex exporting needs, including dumping multiple circuits in a single session, it may be
+more convenient or faster to use the complete interface.
+
+.. autoclass:: Exporter
+    :members:
+
+All of these interfaces will raise :exc:`QASM3ExporterError` on failure.
+
+.. autoexception:: QASM3ExporterError
+
+
+Importing from OpenQASM 3
+=========================
+
+Currently only two high-level functions are offered, as Qiskit support for importing from OpenQASM 3
+is in its infancy, and the implementation is expected to change significantly.  The two functions
+are :func:`load` and :func:`loads`, which are direct counterparts of :func:`dump` and :func:`dumps`,
+respectively loading a program indirectly from a named file and directly from a given string.
+
+.. note::
+
+    While we are still in the initial exploratory release period, to use either function, the
```

Comment:
Really nitpicky: But a newspaper editor might strike "initial" from "initial exploratory release period"

Reply:
Done in fc9931abc.